mbedtls/src/x509/certificate.rs

```diff
@@ -1455,8 +1459,18 @@ cYp0bH/RcPTC0Z+ZaqSWMtfxRrk63MJQF9EXpDCdvQRcTMD9D85DJrMKn8aumq0M
         let mut err = String::new();
         assert_eq!(
             Certificate::verify(&certs, &roots, Some(&mut crl), Some(&mut err)).unwrap_err(),
-            Error::X509CertVerifyFailed
+            codes::X509CertVerifyFailed.into()
         );
         assert_eq!(err, "The certificate has been revoked (is on a CRL)\n");
     }
+
+    #[test]
+    fn check_combined_error_from_sys() {
+        let der = b"\x30\x02\x05\x00";
+        let mut cert = MbedtlsBox::<Certificate>::init().unwrap();
+        let res = unsafe { x509_crt_parse_der((&mut (*cert)).into(), der.as_ptr(), der.len()) }.into_result();
```

Comment:
You should just be able to use `Certificate::from_der` instead of duplicating unsafe code.